src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +46,60 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        if (suitablePeriod(dateFrom, dateTo) == false) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuffer result = new StringBuffer("Отчёт за период " + dateFrom + " - " + dateTo);
+        int salary = 0;
+        for (String retVal: names) {
+            salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String [] q = data[i].split(" ");
```

Comment:
q - not informative naming. when I read a variable's name, I should understand, what's in it

Reply:
oh, I forgot to rename, set for myself until I came up with a name (